### PR TITLE
Cleanup Part 1

### PR DESCRIPTION
**Pull request by neilsong**
Bumped all API dependencies to latest versions (with the exception of passport 0.6 cuz its goofy and doesn't work). Also rewrote some promise handling from .then/.catch to await with types so TS compiles.

Oh and formatted everything in API with eslint-prettier.

Edit 1: Prisma also updated to latest.

### CODE REVIEW

File: api/src/catalog/catalog.d.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/**
+ * @file User type declarations.
+ */
```
Comment (Josh-Cena) [inline api/src/catalog/catalog.d.ts:3]: Don't use `.d.ts` for type declarations—use a normal `.ts` file instead. Also where were these types before? Did they not exist?

**Review by Josh-Cena — CHANGES_REQUESTED** [review]

File: api/src/catalog/catalog.d.ts (hunk 1 above)
Comment (neilsong) [reply api/src/catalog/catalog.d.ts]: Done :)

We can clean up explicitly defined types after you implement the runtime lib.

**Review by neilsong — COMMENTED** [review]
Implemented Sida's type suggestion

**Comment by Josh-Cena** [comment]
Do you know why frontend/yarn.lock is updated? IIUC they are separate projects and bumping one shouldn't update the other

File: api/src/catalog/catalog.utils.ts [hunk 2]
```diff
@@ -9,14 +9,72 @@ import { request } from 'graphql-request';
 
 import winston from '../logging/winston';
 
+/**
+ * Type for Catalog data
+ */
```
Comment (Josh-Cena) [inline api/src/catalog/catalog.utils.ts:14]: Extraneous comment?

File: api/src/catalog/catalog.utils.ts [hunk 3]
```diff
@@ -9,14 +9,72 @@ import { request } from 'graphql-request';
...
+
+export type SeasonsType = {
```
Comment (Josh-Cena) [inline api/src/catalog/catalog.utils.ts:16]: I would recommend removing the `Type` suffix. You wouldn't say `seasonValue` either.

**Review by Josh-Cena — COMMENTED** [review]
Incomplete reviews (the challenge controllers needs a deeper look); just some nits. Overall LGTM

**Comment by maxyuan6717** [comment]
neil i stg if u break coursetable with this

**Comment by neilsong** [comment]
yo @maxyuan6717 chilllll, have some faith in lil bro 🙏

@Josh-Cena, will double check the frontend yarn.lock and the challenge workflow.

File: api/src/catalog/catalog.utils.ts (hunk 3 above)
Comment (neilsong) [reply api/src/catalog/catalog.utils.ts:16]: so just `Seasons` and `Catalog`?

File: api/src/catalog/catalog.utils.ts [hunk 4]
```diff
@@ -9,14 +9,72 @@ import { request } from 'graphql-request';
...
+  seasons: {
+    season_code: string;
+    term: string;
+    year: number;
+  }[];
+};
+
+// type for catalog data
```
[the remark this answers is not in the document]
Comment (neilsong) [reply api/src/catalog/catalog.utils.ts:24]: mhm

**Comment by bearsyankees** [comment]
Looking over this now. @Josh-Cena thank you for also helping with this review. @maxyuan6717 blame neil if it breaks

File: api/src/catalog/catalog.utils.ts (hunk 3 above)
Comment (Josh-Cena) [reply api/src/catalog/catalog.utils.ts:16]: right

**Comment by neilsong** [comment]
Decided to include the prisma migration in this PR since there are no breaking changes with latest.

File: api/src/challenge/challenge.controllers.ts [hunk 5]
```diff
@@ -149,8 +149,10 @@ export const requestChallenge = async (
   // randomly choosing a minimum rating
   const minRating = 1 + Math.random() * 4;
 
+  let evals: requestEvalsQueryResponse;
+  // get a list of all seasons
   try {
-    const evals = await request(GRAPHQL_ENDPOINT, requestEvalsQuery, {
+    evals = await request(GRAPHQL_ENDPOINT, requestEvalsQuery, {
```
Comment (Josh-Cena) [inline api/src/challenge/challenge.controllers.ts:155]: If it's only used inside the block, put it inside the block

File: api/src/challenge/challenge.controllers.ts [hunk 6]
```diff
@@ -291,14 +293,14 @@ export const verifyChallenge = async (
     });
   }
 
+  let true_evals: verifyEvalsQueryResponse;
+  // get a list of all seasons
   try {
-    // check the answers against the true values
-    const trueEvals = await request(GRAPHQL_ENDPOINT, verifyEvalsQuery, {
+    true_evals = await request(GRAPHQL_ENDPOINT, verifyEvalsQuery, {
```
Comment (Josh-Cena) [inline api/src/challenge/challenge.controllers.ts:299]: Use camelCase + move inside the block

**Review by Josh-Cena — APPROVED** [review]
Almost LGTM

**Review by Josh-Cena — APPROVED** [review]
Great!

**Review by bearsyankees — APPROVED** [review]
#CourseTableIsBack

**Pull request merged**